### PR TITLE
Fix Undefined index: assetgroup_id & language_id and batch move to child of self error message

### PR DESCRIPTION
**Pull request by izharaazmi**
Pull Request for Issue #15132.

### Summary of Changes
Fix move to child error message. 
`JObject` was converting exception to string by casting instead of calling `getMessage` on `Exception` object. 
Fixed batch move error message when moving to child of self in menu manager, fields, categories.

### Testing Instructions
See #15132

### Documentation Changes Required
None

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 498eeff798fc8ce34cd69d7b03d3d9cdeaa791e4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15152">issues.joomla.org/tracker/joomla-cms/15152</a>.</sub>

File: libraries/legacy/model/admin.php [hunk 1]
```diff
@@ -255,7 +255,7 @@ public function batch($commands, $pks, $contexts)
 
 		foreach ($this->batch_commands as $identifier => $command)
 		{
-			if (strlen($commands[$identifier]) > 0)
+			if (!empty($commands[$identifier]))
```
Comment (bertmert) [inline libraries/legacy/model/admin.php]: Are you sure? If yes, then we could reduce code and remove definitions of `protected $batch_commands` in models: com_banners/.../banner.php, com_fields/.../field.php, com_fields/.../group.php, com_menus/.../item.php, com_modules/.../module.php, com_tags/../tag.php

**Comment by franz-wohlkoenig** [comment]
is this PR ready for test?

File: libraries/legacy/model/admin.php (hunk 1 above)
Comment (izharaazmi) [inline libraries/legacy/model/admin.php]: @bertmert You're right!

Comment (bertmert) [inline libraries/legacy/model/admin.php]: Thanks. I'll keep in mind to send a PR then that removes these definitions.

Comment (izharaazmi) [inline libraries/legacy/model/admin.php]: This prevents trigger of an unsupported command for the specific model just by passing the matching parameter in the request. Hence listing the supported values in each model is a good idea, imo.

**Comment by bertmert** [comment]
I have tested this item :white_check_mark: successfully on 498eeff798fc8ce34cd69d7b03d3d9cdeaa791e4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15152">issues.joomla.org/tracker/joomla-cms/15152</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**